index-advisor.md

```diff
@@ -0,0 +1,207 @@
+---
+title: Index Advisor
+summary: TiDB Index Advisor.
+---
+
+# Index Advisor Overview
+
+TiDB's Index Advisor helps users optimize their workload by recommending indexes to improve query performance. The new SQL instruction, `RECOMMEND INDEX`, allows users to generate index recommendations for a single query or an entire workload. To avoid the resource-intensive process of physically creating indexes for evaluation, TiDB supports hypothetical indexes—logical indexes that are not materialized. The syntax and usage of Hypo Indexes are detailed in the Appendix.
+
+The Index Advisor analyzes queries to identify indexable columns from relevant clauses (for example, `WHERE`, `GROUP BY`, `ORDER BY`) and generates index candidates. Using the Hypo Index feature, it estimates the performance benefits of these candidates and employs a genetic search algorithm to select the optimal set of indexes. This algorithm begins with single-column indexes and iteratively explores multi-column indexes, leveraging a `What-If` analysis to evaluate potential indexes based on their impact on optimizer plan costs. Indexes are recommended if they reduce the overall cost compared to executing queries without them.
+
+In addition to recommending new indexes, TiDB also offers a feature to suggest dropping inactive indexes, ensuring efficient index management.
+
+# Recommend Index command
+
+The SQL syntax for the new SQL command `Recommend Index` is shown below. The command can be used for a single SQL query (`For` option) or for a workload (`Run` option). The comamnd also supports setting options for subsequent runs of the command.
+
+```sql
+Recommend Index [Run | For <SQL> | <Options>]
+```
+
+## Single Query Option
+
+Below is an example of a single query, assuming 5,000 rows in table  `t` (we omit the insert statements for brevity):
+
+```sql
+mysql> CREATE TABLE t(a int, b int, c int);
+mysql> RECOMMEND INDEX RUN for "select a, b from t where a=1 and b=1";
```

Comment:
Does it command accept "\G" as delimiter?  column-style printing improves readability for single output.

Reply:
Updated.